ollama/_utils.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+import inspect
+from typing import Callable, Union
+
+import pydantic
+from ollama._types import Tool
+
+
+def _parse_docstring(doc_string: Union[str, None]) -> dict[str, str]:
+  parsed_docstring = {'description': ''}
+  if not doc_string:
+    return parsed_docstring
+
+  lowered_doc_string = doc_string.lower()
+
+  if 'args:' not in lowered_doc_string:
+    parsed_docstring['description'] = lowered_doc_string.strip()
+    return parsed_docstring
+
+  else:
+    parsed_docstring['description'] = lowered_doc_string.split('args:')[0].strip()
+    args_section = lowered_doc_string.split('args:')[1]
+
+  if 'returns:' in lowered_doc_string:
+    # Return section can be captured and used
+    args_section = args_section.split('returns:')[0]
+
+  if 'yields:' in lowered_doc_string:
+    args_section = args_section.split('yields:')[0]
+
+  cur_var = None
+  for line in args_section.split('\n'):
```

Comment:
this can be significantly simpler if you start with this

```python
def parse(docstring):
  descriptions = defaultdict(list)

  key = '_'
  for line in docstring.splitlines():
    if line.lower().startswith('args:'):
      key = 'args'
    elif line.lower().startswith('raises:') or line.lower().startswith('returns:') or line.lower().startswith('yields:'):
      key = ''

    if key:
      descriptions[key].append(line.strip())

  return descriptions
```